crates/ruff_python_formatter/src/expression/parentheses.rs

```diff
@@ -141,26 +143,38 @@ impl<'content, 'ast> FormatParenthesized<'content, 'ast> {
     ) -> FormatParenthesized<'content, 'ast> {
         FormatParenthesized { comments, ..self }
     }
+
+    /// Avoid grouping the parenthesized contents.
+    pub(crate) fn with_ungroup(self) -> FormatParenthesized<'content, 'ast> {
+        FormatParenthesized {
+            ungroup: true,
+            ..self
+        }
+    }
 }
 
 impl<'ast> Format<PyFormatContext<'ast>> for FormatParenthesized<'_, 'ast> {
     fn fmt(&self, f: &mut Formatter<PyFormatContext<'ast>>) -> FormatResult<()> {
         let inner = format_with(|f| {
             if self.comments.is_empty() {
-                group(&format_args![
-                    text(self.left),
-                    &soft_block_indent(&Arguments::from(&self.content)),
-                    text(self.right)
-                ])
-                .fmt(f)
+                write!(
+                    f,
+                    [
+                        text(self.left),
+                        &soft_block_indent(&Arguments::from(&self.content)),
+                        text(self.right)
+                    ]
+                )
             } else {
-                group(&format_args![
```

Comment:
In short: we need to avoid this `group` call in _certain_ uses of `parenthesized`. I could add an entirely separate builder if we prefer, but I'll need to add a separately builder for `empty_parenthesized` too.

Reply:
(For `empty_parenthesized`, we could even remove the `group` entirely and require the callers to wrap in `group`. For `parenthesized`, we can't do that, because we wrap the `group` in a `fits_expanded` in the `NodeLevel::Expression` case.)